src/components/TokenSelector.tsx

```diff
@@ -251,9 +252,36 @@ interface Props {
 const TokenSelector: React.FC<Props> = ({ isDisabled, tokens, selected, onChange, tabIndex = 0 }) => {
   const options = useMemo(() => tokens.map(token => ({ token, value: token.symbol, label: token.name })), [tokens])
 
+  // isFocused is used to force the menu to remain open and give focus to the search input
+  const [isFocused, setIsFocused] = useState(false)
+
+  const onSelectChange = useCallback(
+    (selected: { token: TokenDetails }, { action }: ActionMeta): void => {
+      // When an option is chosen, give control back to react-select
+      setIsFocused(false)
+
+      if (selected && action === 'select-option' && 'token' in selected) {
+        onChange(selected.token)
+      }
+    },
+    [onChange],
+  )
+
+  // When the search input is focused, force menu to remain open
+  const onMenuInputFocus = useCallback(() => setIsFocused(true), [])
```

Comment:
is this callBack use necessary? there are no deps

Reply:
Exactly. We don't need this to be updated at all. Otherwise the function would be re-instantiated at every render. No?